### PR TITLE
test: characterize truncateToolResult (BAT-1077, #345 Phase 0)

### PR DESCRIPTION
**Pull request by sepivip**
## Summary
Characterization test pinning the **current** behavior of `truncateToolResult` (`app/src/main/assets/nodejs-project/config.js`) — the formatter applied to **every tool result** before it enters the conversation (sole consumer: `ai.js` `truncateToolResult(JSON.stringify(result))`).

This is **Phase 0, item (e)** (tool-result-formatting half) of #345. The audit found this is the single biggest *uncovered* gap: the only existing test that touches the function stubs it as an identity no-op, so a truncation regression ships uncaught. Pinning it now is the safety net for the eventual `ai.js` decomposition.

**Test-only — zero production code touched. RC-safe.** Contract: BAT-1077 (PM/Codex-approved).

## What's here
- New `tests/nodejs-project/truncate-tool-result.test.js` (19 cases).
- One line into the `build.yml` CI for-loop allowlist (above `ci-coverage-manifest`, preserving the no-trailing-backslash final entry; `ci-coverage-manifest.test.js` enforces classification).

## Approach
Can't `require('config.js')` directly (it reads a real `config.json` and `process.exit(1)`s on load), so — matching the `env-merge`/`active-model` convention — the test (1) mirrors the function verbatim and asserts behavior, and (2) drift-guards the live source with **comment-stripped structural regexes**. Exact marker bytes (⚠️ / em-dash / leading `\n\n`) are pinned as a **behavioral assertion on output** (independent golden literal), with tolerant semantic anchors on the source.

## Coverage
Non-string passthrough (incl. `NaN`); `<=50000` unchanged (49999 / exactly 50000); hard cut at 50000 (no newline / newline-only-`<2000` / newline-beyond-cap); line-boundary selection (newline at 49000; latest-of-multiple wins); **`MIN_KEEP_CHARS` boundary — index 2000 accepted, 1999 falls back**; byte-exact marker; `droppedChars` excludes marker; returned length exceeds cap; CRLF `\r` retained; UTF-16 surrogate split; structural source drift-guard.

## Test plan
- [x] `node tests/nodejs-project/truncate-tool-result.test.js` — **19/19 pass**
- [x] `node tests/nodejs-project/smoke.js` — **PASS** (78/78 parse, 27/27 load)
- [x] `node tests/nodejs-project/ci-coverage-manifest.test.js` — **5/5 pass** (allowlist parses; new file classified; not double-listed)
- [x] `scripts/pre-push-check.sh` — **ALL CHECKS PASSED** (Kotlin dappStoreDebug compile + Node smoke + tool schemas + wallets prompt)

## Self-Awareness (SAB)
N/A — does not touch `buildSystemBlocks()`, `DIAGNOSTICS.md`, JS error/warn log sites, or any user-visible AI capability. No Telegram commands added.

## Known limitations
- Characterizes `truncateToolResult` only; the context-trimming half of #345 item (e) (`adaptiveTrim`/`summarizeOldMessages`) is a deliberate follow-up.
- The mirror is a manual copy; the source drift-guard catches structural divergence but (by design) tolerates marker whitespace reflow.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Expanded CI to run an additional Node.js truncation scenario test as part of the existing “Financial-flow + confirmation-gate tests” suite.
  * Added a comprehensive truncation behavior test set covering boundary conditions, newline/CRLF handling, marker formatting, and Unicode edge cases to improve confidence in output correctness.

### CODE REVIEW

**Comment by sepivip** [comment]
@coderabbitai full review

File: tests/nodejs-project/truncate-tool-result.test.js [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    // line-boundary selection + MIN_KEEP_CHARS fallback
+    assert.ok(/lastIndexOf\s*\(\s*['"]\\n['"]\s*,/.test(code),
+        "lastIndexOf('\\n', ...) line-boundary call missing");
+    assert.ok(/cutoff\s*<\s*MIN_KEEP_CHARS/.test(code),
+        'cutoff < MIN_KEEP_CHARS fallback branch missing');
+    // tolerant marker anchors (semantic, reflow-safe) — exact bytes pinned on output above
+    assert.ok(/Content truncated/.test(code), 'marker "Content truncated" wording missing');
+    assert.ok(/characters removed/.test(code), 'marker "characters removed" wording missing');
+    assert.ok(/offset\/limit/.test(code), 'marker "offset/limit" recovery hint missing');
+});
```
[the remark this answers is not in the document]
Comment (sepivip) [reply tests/nodejs-project/truncate-tool-result.test.js]: Fixed in ac07e41f. Replaced the hand-copied mirror with an isolated `vm` harness that extracts and executes the **live** `truncateToolResult` (+ its 4 constants) from `config.js` source, so all 19 behavioral cases now assert against the real function — a semantic change in `config.js` is caught, not silently mirrored. Verified: a `<` → `<=` mutation of the `MIN_KEEP_CHARS` fallback flips the newline@2000 cutoff 2000 → 50000 and fails the test (the old mirror could not). A slim guard remains only to confirm `truncateToolResult` is still exported from `module.exports` (the vm runs the body regardless of export, and ai.js imports it). The ast-grep path note is a false positive — `CONFIG_JS` is a build-time constant path, same as env-merge.test.js. Thanks for the catch.

**Comment by sepivip** [comment]
@coderabbitai review
